cms/core/tests/test_video_embed_block.py

```diff
@@ -0,0 +1,302 @@
+from django.test import TestCase
+from wagtail.blocks import StructBlockValidationError
+from wagtail_factories import ImageFactory
+
+from cms.core.blocks import VideoEmbedBlock
+
+
+class VideoEmbedBlockTestCase(TestCase):
+    """Test for video embed block."""
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.video_block = VideoEmbedBlock()
+        cls.image = ImageFactory.create()
+
+    def test_clean__invalid_domain(self):
+        """Check the VideoEmbedBlock validates the supplied URL."""
+        with self.assertRaises(StructBlockValidationError) as info:
+            value = self.video_block.to_python(
+                {
+                    "link_url": "https://ons.gov.uk/908205163",
+                    "image": self.image.id,
+                    "title": "The video",
+                    "link_text": "Watch the video",
+                }
+            )
+            self.video_block.clean(value)
+
+        self.assertEqual(
+            info.exception.block_errors["link_url"].message,
+            "The link URL must use a valid Vimeo or YouTube video URL",
+        )
+
+    def test_clean__invalid_youtube_link_1(self):
```

Comment:
Non-blocking: A lot of the unit tests here are repeated patterns, perhaps we could use [subtests](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests) to parameterise them.

Reply:
Thanks @AdamHawtin, @zerolab made the same observation, but suggested it could be follow-up work.

Reply:
Do we want a ticket for this? @helenb

Reply:
Good point, I will create one.

Reply:
Done: https://jira.ons.gov.uk/browse/CMS-297